devdoc/channel_2.md

```diff
@@ -0,0 +1,72 @@
+# channel_2 requirements:
+
+## Overview
+
+The current implementation of `channel` has certain pitfalls. `channel_2` aims to address these pitfalls by making the following changes:
+
+1. Introduce a `channel_2_reset` API: This API will allow the user to reset the `channel_2` to a clean state. It will cause all pending operations to be abandoned.
```

Comment:
>channel_2_reset [](http://example.com/codeflow?start=16&length=15)

This reset smells like close and then open again.
I do have to wonder why it is not modelled as channel having an open/closed state and then the user can do whatever they want to "reset"